core/src/Borrower.sol

```diff
@@ -286,11 +292,19 @@ contract Borrower is IUniswapV3MintCallback {
      * TWAPs. If any of the highest 8 bits are set, we fallback to binary search.
      */
     function modify(IManager callee, bytes calldata data, uint40 oracleSeed) external payable {
-        require(_loadSlot0() % (1 << 160) == uint160(msg.sender), "Aloe: only owner");
+        uint256 slot0_ = _loadSlot0();
+        // Equivalent to `slot0.state == State.Ready`
+        require(slot0_ >> 248 == uint256(State.Ready) && msg.sender == owner(), "Aloe: only owner");
 
-        _saveSlot0(uint160(msg.sender), _formatted(State.InModifyCallback));
-        int24[] memory positions_ = positions.write(callee.callback(data, msg.sender));
-        _saveSlot0(uint160(msg.sender), _formatted(State.Ready));
+        _saveSlot0(slot0_, State.InModifyCallback);
+        // Hand control to `callee` and accept new `positions` if it returned any
+        uint144 positions = callee.callback(data, msg.sender, uint144(slot0_));
+        assembly ("memory-safe") {
+            // Equivalent to `if (positions > 0) slot0_ = positions`
+            slot0_ := or(positions, mul(slot0_, iszero(positions)))
+        }
```

Comment:
I'm not sure whether to keep this or not. Let me know what yall think.

Pro:
- `IManager`s can still return 0 if they don't want to update `positions`

Con:
- Costs a few extra gas and makes the code messy here
- Also, since we pass the current `uint144 positions` to the callback anyway, it wouldn't be that hard for `IManager`s to return it unchanged (instead of returning 0)

Reply:
Update: I decided to remove it.

Reply:
off with it